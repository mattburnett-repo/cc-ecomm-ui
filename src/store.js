import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;