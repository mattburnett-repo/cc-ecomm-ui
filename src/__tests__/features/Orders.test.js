
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
// import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';
import store from '../../store.js'

import Orders from '../../features/orders/orders';

describe('<Orders /> feature tests', () => {
    beforeEach(() => render(<Provider store={store}> <Orders/></Provider>));

    it('should render an OrdersDiplay component', () => {
        screen.getByRole('presentation', { name: /^orders$/i})
    })

    // it('should render order items from data', () => {
    //     const theVals = screen.getAllByRole('presentation', { name: /^order$/i})

    //     expect(theVals).toHaveLength(1)
    //     expect(theVals[0]).toHaveTextContent('order_id')
    //     expect(theVals[0]).toHaveTextContent('user_id')
    //     expect(theVals[0]).toHaveTextContent('order_date')
    //     expect(theVals[0]).toHaveTextContent('total_price')
    // })

    // it('creates a snapshot', () => {
    //     const tree = renderer.create(<Provider store={store}> <Orders/></Provider>).toJSON();
    //     expect(tree).toMatchSnapshot();  
    // });
})