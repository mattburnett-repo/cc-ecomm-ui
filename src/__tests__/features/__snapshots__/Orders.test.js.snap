// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Orders /> feature tests creates a snapshot 1`] = `
Array [
  " ",
  <div
    aria-label="orders"
    role="presentation"
  >
    <h2>
      OrdersDisplay
    </h2>
  </div>,
]
`;
